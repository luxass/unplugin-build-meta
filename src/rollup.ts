import { createRollupPlugin } from "unplugin";
import { unpluginFactory } from "./";

/**
 * Rollup plugin
 *
 * @example
 * ```ts
 * // rollup.config.js
 * import buildMeta from "unplugin-build-meta/rollup"
 *
 * export default {
 *   plugins: [buildMeta()],
 * }
 * ```
 */
export default createRollupPlugin(unpluginFactory);
