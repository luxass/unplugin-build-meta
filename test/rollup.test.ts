import { join } from "node:path";
import { rollup } from "rollup";
import { describe, expect, it } from "vitest";
import { testdir } from "vitest-testdirs";
import buildMeta from "../src/rollup";

describe("handles git metadata", () => {
  it("expect git metadata to be available", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/git"));

    expect(testdirPath).toBeDefined();

    const bundle = await rollup({
      input: join(testdirPath, "git.js"),
      plugins: [buildMeta({
        modules: ["git"],
      })],
    });

    const { output } = await bundle.generate({
      format: "es",
    });

    expect(output).toBeDefined();
    expect(output[0]).toBeDefined();
    expect(output[0].code).toBeDefined();
    expect(output[0].code).toMatchSnapshot();
  });

  it("expect specific git properties to be importable", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/git"));

    expect(testdirPath).toBeDefined();

    const bundle = await rollup({
      input: join(testdirPath, "git-specific.js"),
      plugins: [buildMeta({
        modules: ["git"],
      })],
    });

    const { output } = await bundle.generate({
      format: "es",
    });

    expect(output).toBeDefined();
    expect(output[0]).toBeDefined();
    expect(output[0].code).toBeDefined();
    expect(output[0].code).toMatchSnapshot();
  });
});
