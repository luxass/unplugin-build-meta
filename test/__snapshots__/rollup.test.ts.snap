// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`handles git metadata > expect git metadata to be available 1`] = `
"const repositoryUrl = "https://github.com/luxass/unplugin-build-meta";
const sha = "0d26380e72f1cc40efa0388abe3e2e998db9d772";
const shortSha = "0d26380e72";
const branch = "main";
const tag = null;
const tags = [];
const lastTag = null;
const commitAuthorName = "Lucas";
const commitAuthorEmail = "lucasnrgaard@gmail.com";
const commitAuthorDate = "2025-04-16T16:53:03+02:00";
const commitCommitterName = "Lucas";
const commitCommitterEmail = "lucasnrgaard@gmail.com";
const commitCommitterDate = "Wed Apr 16 16:53:03 2025 +0200";
const latestCommitMessage = "chore: fix eslint ignore";

var git = /*#__PURE__*/Object.freeze({
	__proto__: null,
	branch: branch,
	commitAuthorDate: commitAuthorDate,
	commitAuthorEmail: commitAuthorEmail,
	commitAuthorName: commitAuthorName,
	commitCommitterDate: commitCommitterDate,
	commitCommitterEmail: commitCommitterEmail,
	commitCommitterName: commitCommitterName,
	lastTag: lastTag,
	latestCommitMessage: latestCommitMessage,
	repositoryUrl: repositoryUrl,
	sha: sha,
	shortSha: shortSha,
	tag: tag,
	tags: tags
});

console.log(git);
"
`;

exports[`handles git metadata > expect specific git properties to be importable 1`] = `
"const sha = "0d26380e72f1cc40efa0388abe3e2e998db9d772";
const shortSha = "0d26380e72";
const branch = "main";

console.log({ branch, sha, shortSha });
"
`;
